src/Education/Rule/SLCoreRuleMetaDataProvider.cs

```diff
@@ -74,7 +72,35 @@ public async Task<IRuleInfo> GetRuleInfoAsync(SonarCompositeRuleId ruleId)
         return null;
     }
 
-    private static RuleInfo Convert(EffectiveRuleDetailsDto effectiveRuleDetailsAsync) =>
+    /// <inheritdoc />
+    public async Task<IRuleInfo> GetRuleInfoAsync(SonarCompositeRuleId ruleId, Guid? issueId)
+    {
+        var ruleInfoFromIssue = issueId != null ? await GetEffectiveIssueDetailsAsync(issueId.Value) : null;
+
+        return ruleInfoFromIssue ?? await GetRuleInfoAsync(ruleId);
+    }
+
+    internal async Task<IRuleInfo> GetEffectiveIssueDetailsAsync(Guid issueId)
```

Comment:
This should belong to the main interface instead

Reply:
Why? I designed the contact to expose the GetRuleInfoAsync(SonarCompositeRuleId ruleId, Guid? issueId), because in many of the cases the issued is null. In which case, we should still provide the ruleInfo and not let the callers deal with this case.

Reply:
As discussed, I only exposed one method in the IRuleMetadataProvider